File: src/main/java/net/hectus/neobb/turn/person_game/warp/PTAmethystWarp.java
package net.hectus.neobb.turn.person_game.warp;

import net.hectus.neobb.player.NeoPlayer;
import net.hectus.neobb.structure.PlacedStructure;
import org.bukkit.World;

public class PTAmethystWarp extends PWarpTurn {
    public PTAmethystWarp(World world) { super(world, "amethyst"); }
    public PTAmethystWarp(PlacedStructure data, World world, NeoPlayer player) { super(data, world, player, "amethyst"); }

    @Override
    public void apply() {
        super.apply();
        player.health(player.game.info().startingHealth());
    }

    @Override
    public double damage() {
        return 1.0;
    }

    @Override
    public int chance() {
        return 10;
    }
}

File: src/main/kotlin/net/hectus/neobb/turn/person_game/warp/PTAmethystWarp.kt
package net.hectus.neobb.turn.person_game.warp

import com.marcpg.libpg.storing.Cord
import net.hectus.neobb.player.NeoPlayer
import net.hectus.neobb.structure.PlacedStructure
import net.hectus.neobb.structure.StaticStructure
import net.hectus.neobb.structure.StaticStructures

class PTAmethystWarp(data: PlacedStructure?, cord: Cord, player: NeoPlayer?) : PWarpTurn(data, cord, "amethyst", player) {
    override val staticStructure: StaticStructure = StaticStructures.Person.Warp.AMETHYST

    override val damage: Double = 1.0
    override val chance: Double = 10.0

    override fun apply() {
        player!!.health(player.game.info.startingHealth)
    }
}
